src/torrent.ts

```diff
@@ -201,8 +201,32 @@ export async function getTorrentByFuzzyName(
 	name: string
 ): Promise<null | Metafile> {
 	const allNames = await db("torrent").select("name", "file_path");
+
+	const episode = name.match(EP_REGEX);
+	const season = name.match(SEASON_REGEX);
+	const movie = name.match(MOVIE_REGEX);
+	const fullMatch = (episode?.[0] ?? season?.[0] ?? movie?.[0])
+		.replaceAll(".", "").replaceAll(" ", "").toLowerCase();
+
+	// Attempt to filter torrents in DB to match incoming torrent before fuzzy check
+	var filteredNames = [];
+
+	if (fullMatch) {
+		filteredNames = allNames.filter((dbName) => {
+			const dbEpisode = dbName.match(EP_REGEX);
+			const dbSeason = dbName.match(SEASON_REGEX);
+			const dbMovie = dbName.match(MOVIE_REGEX);
+			const dbMatch = (dbEpisode?.[0] ?? dbSeason?.[0] ?? dbMovie?.[0])
+			.replaceAll(".", "").replaceAll(" ", "").toLowerCase();
+			if (!dbMatch) return false;
+			return fullMatch === dbMatch});
+	}
+
+	// If none match, proceed with fuzzy name check on all names.
+
 	// @ts-expect-error fuse types are confused
-	const potentialMatches = new Fuse(allNames, {
+	const potentialMatches = new Fuse(filteredNames.length > 0 ? filteredNames : allNames, 
```

Comment:
I read about [uFuzzy](https://github.com/leeoniya) a while back, which has been on my list to replace Fuse with (every fuse search uses a ridiculous amount of CPU and locks the main thread for a while. if you're interested, this could be a pretty nice "while you're here" free performance improvement

Reply:
I'll take a look

Reply:
👋 ping me if you have questions 😉 

https://github.com/leeoniya/uFuzzy/releases/tag/1.0.10